phoenix-spark/src/main/scala/org/apache/phoenix/spark/ProductRDDFunctions.scala

```diff
@@ -16,19 +16,20 @@ package org.apache.phoenix.spark
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.io.NullWritable
 import org.apache.phoenix.mapreduce.PhoenixOutputFormat
-import org.apache.phoenix.mapreduce.util.{ColumnInfoToStringEncoderDecoder, PhoenixConfigurationUtil}
+import org.apache.phoenix.mapreduce.util.PhoenixConfigurationUtil
 import org.apache.spark.Logging
 import org.apache.spark.rdd.RDD
+
 import scala.collection.JavaConversions._
 
 class ProductRDDFunctions[A <: Product](data: RDD[A]) extends Logging with Serializable {
 
   def saveToPhoenix(tableName: String, cols: Seq[String],
-                    conf: Configuration = new Configuration, zkUrl: Option[String] = None)
+                    conf: Configuration = new Configuration, zkUrl: Option[String] = None, tenantId: Option[String] = None)
```

Comment:
Looking at this closer it appears the only way to pass the tenantId to PairRDDFunctions.saveAsNewAPIHadoopFile is through the Configuration parameter. 

I was able to avoid setting TenantId in the partitionConfig by overloading getUpsertColumnMetadataList to take in a Properties object. However PairRDDFunctions is part of spark-core, so I don't know if we want to make a change to spark-core so we can avoid setting the tenantId in the outputConfig. 

Let me know what you think.